app/controllers/search_controller.rb

```diff
@@ -5,6 +5,7 @@ def index
     @form = SearchForm.new(search_form_params)
     if @form.valid?
       @result = FirmRepository.new.search(@form.to_query, page: page)
+      redirect_to "/404.html" if @result.total_pages < page
```

Comment:
If the page returns a 404 status code, that will stop Google indexing it, because you are declaring "this isn't a valid page" rather than "what you're looking for is actually over there".